gameClient.js

```diff
@@ -645,19 +645,20 @@
   function emitMessage(msg) {
     if (msg.length > 0) {
       socket.emit('msgSent', {room: g_roomID, message: msg.slice(0, 150), senderName: g_clientPlayer.name});
-      $('#chatInput').val('');
     }
   }
 
-  $('.textEntry').submit(function(e){
+  $('.textEntry').on('submit', function(e) {
     e.preventDefault(); // prevents page reloading
     emitMessage($('#chatInput').val());
+    $('#chatInput').val('');
     return false;
   });
 
-  $('#chatInput').keydown(function() {
-    if (event.keyCode == 13) { //enter
-      emitMessage($('#chatInput').val());
+  $('#chatInput').on('keydown', function(e) {
+    if (e.key === 'Enter') {
+      e.preventDefault(); // prevents newline
+      $('.textEntry').trigger('submit');
```

Comment:
- `.submit()` and `.keydown()` are deprecated
- `event.keyCode` is deprecated
- The `emitMessage` function shouldn't be responsible for clearing the `chatInput`